packages/workspace-store/src/client.ts

```diff
@@ -9,14 +9,33 @@ import { extensions } from '@/schemas/extensions'
 import { reactive } from 'vue'
 import { coerceValue } from '@/schemas/typebox-coerce'
 import { OpenAPIDocumentSchema } from '@/schemas/v3.1/strict/openapi-document'
+import { defaultReferenceConfig, type ReferenceConfig } from '@/schemas/reference-config'
+import { mergeObjects } from '@/helpers/merge-object'
+import type { DeepTransform } from '@/types'
 
-type WorkspaceDocumentMetaInput = { meta?: WorkspaceDocumentMeta; name: string; config?: createNavigationOptions }
+// TODO: merge navigation options with the document config
+type WorkspaceDocumentMetaInput = {
+  meta?: WorkspaceDocumentMeta
+  name: string
+  config?: Config & Partial<createNavigationOptions>
+}
 
 type UrlDoc = { url: string } & WorkspaceDocumentMetaInput
 type ObjectDoc = { document: Record<string, unknown> } & WorkspaceDocumentMetaInput
 
 type WorkspaceDocumentInput = UrlDoc | ObjectDoc
 
+/**
+ * Configuration object for workspace documents.
+ */
+type Config = {
+  'x-scalar-reference-config'?: ReferenceConfig
+}
+
+const defaultConfig: DeepTransform<Config, 'NonNullable'> = {
+  'x-scalar-reference-config': defaultReferenceConfig,
+}
```

Comment:
let’s stay closer to the package name

```suggestion
type Config = {
  'x-scalar-reference-config'?: ReferenceConfig
}

const defaultConfig: DeepTransform<Config, 'NonNullable'> = {
  'x-scalar-api-reference': defaultApiReferenceConfiguration,
}
```